DICOM/src/Microsoft.Health.DeID.SharedLib/CryptoHashFunction.cs

```diff
@@ -8,67 +8,64 @@
 using System.Security.Cryptography;
 using System.Text;
 using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
 
 namespace Microsoft.Health.Dicom.DeID.SharedLib
 {
     public class CryptoHashFunction
     {
-        public static byte[] ComputeHmacSHA256Hash(byte[] input, byte[] hashKey = null)
+        private readonly HMAC _hmac;
+
+        public CryptoHashFunction(CryptoHashSetting cryptoHashSetting = null)
+        {
+            cryptoHashSetting ??= new CryptoHashSetting();
+            _hmac = cryptoHashSetting.CryptoHashKey == null ? new HMACSHA256() : new HMACSHA256(Encoding.UTF8.GetBytes(cryptoHashSetting.CryptoHashKey));
+        }
+
+        public byte[] ComputeHmacSHA256Hash(byte[] input)
         {
             if (input == null)
             {
                 return input;
             }
 
-            HMAC hmac = new HMACSHA256();
-            if (hashKey != null)
-            {
-                hmac = new HMACSHA256(hashKey);
-            }
-
-            return hmac.ComputeHash(input);
+            return _hmac.ComputeHash(input);
         }
 
-        public static byte[] ComputeHmacSHA256Hash(Stream input, byte[] hashKey = null)
+        public byte[] ComputeHmacSHA256Hash(Stream input)
         {
             if (input == null)
             {
                 return null;
             }
 
-            HMAC hmac = new HMACSHA256();
-            if (hashKey != null)
-            {
-                hmac = new HMACSHA256(hashKey);
-            }
-
-            return hmac.ComputeHash(input);
+            return _hmac.ComputeHash(input);
         }
 
-        public static byte[] ComputeHmacSHA256Hash(string input, byte[] hashKey = null, Encoding encoding = null)
+        public byte[] ComputeHmacSHA256Hash(string input, Encoding encoding = null)
         {
             if (input == null)
             {
                 return null;
             }
 
             encoding ??= Encoding.UTF8;
-            return ComputeHmacSHA256Hash(encoding.GetBytes(input), hashKey);
+            return ComputeHmacSHA256Hash(encoding.GetBytes(input));
```

Comment:
>GetBytes [](http://example.com/codeflow?start=50&length=8)

In general this could throw exceptions.
Especially when the input text comes from user data.
Will the caller catch the exception?